be/src/exec/pipeline/analysis/analytic_sink_operator.cpp

```diff
@@ -273,6 +283,37 @@ void AnalyticSinkOperator::_process_by_partition_for_unbounded_preceding_rows_fr
     }
 }
 
+void AnalyticSinkOperator::_process_by_partition_for_unbounded_preceding_range_frame_materializing(
+        size_t chunk_size, bool is_new_partition) {
+    if (_analytor->has_cume_function()) {
+        _analytor->set_partition_size_for_cume_function();
+    }
+    do {
```

Comment:
I'm somewhat perplexed. According to the function `_process_by_partition_if_necessary_materializing`, computations are only performed when all data within a partition have arrived. Additionally, when calculating rank, I'm using `state.count = _partition_end - _partition_start;` instead of `chunk_size`. Would this approach potentially be accurate?

```c++
Status AnalyticSinkOperator::_process_by_partition_if_necessary_materializing() {
    while (_analytor->has_output()) {
        ...
        _analytor->find_partition_end();
        // Only process after all the data in a partition is reached
        if (!_analytor->found_partition_end().first) {
            return Status::OK();
        }

        const auto chunk_size = static_cast<int64_t>(_analytor->current_chunk_size());
        _analytor->create_agg_result_columns(chunk_size);

        bool is_new_partition = _analytor->is_new_partition();
        if (is_new_partition) {
            _analytor->reset_state_for_cur_partition();
        }

        (this->*_process_by_partition)(chunk_size, is_new_partition);

        // Chunk may contains multiply partitions, so the chunk need to be reprocessed
        if (_analytor->is_current_chunk_finished_eval(chunk_size)) {
            ChunkPtr chunk;
            RETURN_IF_ERROR(_analytor->output_result_chunk(&chunk));
            _analytor->offer_chunk_to_buffer(chunk);
        }
    }
    return Status::OK();
}
```